encoding.go

```diff
@@ -86,21 +94,31 @@ var (
 // initEncoding initializes all the decoder/encoder pools with the Content-Types found
 // in JSONContentTypes and GobContentTypes. JSON is set as the default decoder.
 func (app *Application) initEncoding() {
+	contentTypeCount := len(JSONContentTypes) + len(XMLContentTypes) + len(GobContentTypes)
+	decoders := make(map[string]*decoderPool, contentTypeCount)
+	encoders := make(map[string]*encoderPool, contentTypeCount)
+
+	// Add json support
 	jf := &jsonFactory{}
 	dp := newDecodePool(jf)
 	ep := newEncodePool(jf)
 	app.defaultDecoderPool = dp
 	app.defaultEncoderPool = ep
-
-	contentTypeCount := len(JSONContentTypes) + len(GobContentTypes)
-	decoders := make(map[string]*decoderPool, contentTypeCount)
-	encoders := make(map[string]*encoderPool, contentTypeCount)
-
 	for _, contentType := range JSONContentTypes {
 		decoders[contentType] = dp
 		encoders[contentType] = ep
 	}
 
+	// Add xml support
+	xf := &xmlFactory{}
+	dp = newDecodePool(xf)
+	ep = newEncodePool(xf)
+	for _, contentType := range GobContentTypes {
```

Comment:
Is this correct? shouldn't it be adding XMLContentTypes decoders and encoders here and Gob ones down below?